views/js/pciManager/pciManager.js

```diff
@@ -115,68 +116,106 @@ define([
                 $uploader.uploader('reset');
                 hider.show($uploader);
             })
-            .on('updateListing', function(){
+            .on('updateListing', function () {
                 var self = this,
-                    urls = _.pick(this.config, ['disableUrl', 'enableUrl']),
+                    urls = _.pick(this.config, ['disableUrl', 'enableUrl', 'unregisterUrl', 'exportPciUrl']),
                     $fileSelector = this.getElement().find('.file-selector'),
                     $fileContainer = $fileSelector.find('.files'),
                     $placeholder = $fileSelector.find('.empty');
-
-                if(_.size(listing)){
+                if (_.size(listing)) {
 
                     hider.hide($placeholder);
 
                     $fileContainer
                         .empty()
                         .html(listingTpl({
-                            interactions : listing
+                            interactions: listing
                         }));
 
-                    $fileContainer.find('.switch-box').each(function(){
+                    $fileContainer.find('.switch-box').each(function () {
```

Comment:
So instead of doing 3 `_.each` loops back-to-back, I would wrap everything into one loop.
```
$fileContainer.find('.actions').each(function () {
    var $actions = $(this);
    var $switch = $(this).find('.switch-box');
    var $li = $switch.closest('.pci-list-element');
    var typeIdentifier = $li.data('type-identifier');

    switchFactory($switch, {
    });

    buttonFactory({
    });

    buttonFactory({
    });
});
```